File: frontend/lib/model/order.dart
import 'package:frontend/model/enums/order_status.dart';
import 'package:frontend/model/order_item.dart';

class Order {
  final String orderId, userId, userFullName;
  final OrderStatus status;
  final DateTime createdAt;
  final List<OrderItem> items;

  Order({
    required this.orderId,
    required this.userId,
    required this.userFullName,
    required this.status,
    required this.createdAt,
    required this.items,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    final itemsJson = json['items'] as List<dynamic>? ?? [];
    return Order(
      orderId: json['orderId'] as String,
      userId: json['userId'] as String,
      userFullName: json['userFullName'] as String? ?? '',
      status: OrderStatusExtension.fromBackend(json['status'] as String),
      createdAt: (DateTime.parse(json['createdAt'] as String)).toLocal(),
      items: itemsJson
          .map((item) => OrderItem.fromJson(item as Map<String, dynamic>))
          .toList(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "orderId": orderId,
      "userId": userId,
      "userFullName": userFullName,
      "status": status.toBackend(),
      "createdAt": createdAt.toUtc().toIso8601String(),
      "items": items.map((item) => item.toJson()).toList(),
    };
  }
}

File: dashboard/src/model/order.ts
import { createOrderItems, OrderItem, RawOrderItem } from "./order_item";

export type OrderStatus = "PLACED" | "PREPARING" | "READY" | "COMPLETED";

export interface RawOrder {
  orderId: string;
  userId: string;
  status: OrderStatus;
  userFullName: string;
  createdAt: string;
  items: RawOrderItem[];
}

export interface Order {
  orderId: string;
  status: OrderStatus;
  userFullName: string;
  createdAt: string;
  items: OrderItem[];
  placedAt: string;
  customerName: string;
}

export function createOrder(raw: RawOrder): Order {
  return {
    orderId: raw.orderId,
    status: raw.status,
    userFullName: raw.userFullName,
    createdAt: raw.createdAt,
    items: createOrderItems(raw.items),
    placedAt: raw.createdAt,
    customerName: raw.userFullName,
  };
}